content/blog/nginx-ingress-retirement-what-you-need-to-know.md

```diff
@@ -0,0 +1,154 @@
+
+In Kubernetes, an **[Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/)** is essential—it watches Ingress objects in the cluster and programs NGINX accordingly, routing incoming traffic to the applications in your cluster.
+
+Here’s a simple example of an Ingress object:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example-ingress
+  namespace: default
+spec:
+  rules:
+  - host: example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: example-service
+            port:
+              number: 80
+```
+
+This config routes traffic from `example.com` to the backend service `example-service` on port 80, allowing users to reach your application from outside the cluster.
+
+## Why Gateway API?
+
+One challenge with the Ingress API is **inconsistent behavior across vendors**, largely due to its reliance on annotations. These overloaded annotations are project-specific and can behave unpredictably when migrating between implementations. Earlier this year, Wiz Research disclosed [several CVEs](https://www.wiz.io/blog/ingress-nginx-kubernetes-vulnerabilities) in NGINX related to annotation-based authentication or UID configuration.
+
+Another challenge is the **lack of a proper status field**, which makes troubleshooting difficult. While you can inspect rules, events, and annotations, it’s often unclear why an Ingress isn’t working.
+
+Because of these limitations, the Kubernetes community has been evolving the Ingress API since KubeCon 2019. The Gateway API reached **[GA](https://kubernetes.io/blog/2023/10/31/gateway-api-ga/)** in **October 2023**, making core APIs like `Gateway`, `GatewayClass`, and `HTTPRoute` stable.
+
+## Advantages of Gateway API
+
+What we love about the Gateway API:
+- Extensibility: You can define custom traffic policies, rate limits, and more.
+- Status field: Provides clear insights into whether a resource is accepted, programmed, and error-free.
+
+Example with **kgateway** using the core Gateway and the extended TrafficPolicy resources:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: http
+spec:
+  gatewayClassName: kgateway
+  listeners:
+  - protocol: HTTP
+    port: 80
+    name: http
+---
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: transformation
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: http
+  transformation:
+    request:  
+      add:
+      - name: x-forwarded-uri
+        value: 'https://{{ request_header(":authority") }}{{ request_header(":path") }}'
+```
+
+After deployment, the status field for each resource lets you see whether resources are programmed correctly and if any errors exist. Here’s an example of the status field for the `http` Gateway resource:
```

Comment:
```suggestion
After deployment, the status field for each resource lets you see whether resources are programmed correctly and if any errors exist. Here’s an example of the status field for the TrafficPolicy resource:
```

Reply:
Thanks for catching this, I've updated the status example you provided to me :)